lang/stack/Portfile

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github                      1.0
+
+github.setup        commercialhaskell stack 2.1.1
+
+name                stack
+categories          lang haskell
+maintainers         nomaintainer
+license             BSD
+platforms           darwin
+
+description The Haskell Tool Stack
+long_description    ${description}. \
+    Stack is a cross-platform program for developing Haskell projects. \
+    It is intended for Haskellers both new and experienced.
+
+homepage            http://haskellstack.org
+
+variant bootstrap \
+	    description {Bootstrap a build of stack.} {}
+
+if {[variant_isset "bootstrap"]} {
+    pre-fetch {
+        master_sites    ${github.homepage}/archive/v${github.version}
+        checksums       rmd160  60f7fcc5548aabec357be217d11d0d1d616c1806 \
+                        sha256  10c0119456748b189724ee42dea093c324d101487c3d75d57eb625675bb57424 \
+                        size    816217
+    }
+} else {
+    fetch {}
+    checksum {}
+    extract.cmd /usr/bin/true
+    extract.mkdir yes
+}
+
+use_configure       no
+
+# stack must bootstrap itself
+# See: https://github.com/commercialhaskell/stack/blob/master/doc/README.md
+pre-build {
+    system -W ${worksrcpath} "/bin/mkdir ./bin && /usr/bin/curl -sSL https://get.haskellstack.org/ | /bin/sh -s - -d ./bin"
```

Comment:
Doing curl into sh without checksumming what's fetched seems very bad. I think normal users would not expect something like that to occur in macports.

Reply:
Yeah, that’s what the standard install is, but still…

I changed this to a direct download from the GitHub release page with a checksum.